homeassistant/components/wiz/light.py

```diff
@@ -16,83 +17,178 @@
+            bulb_type.kelvin_range.max
+        ), color_utils.color_temperature_kelvin_to_mired(bulb_type.kelvin_range.min)
+    except WizLightNotKnownBulb:
+        _LOGGER.debug("Kelvin is not present in the library. Fallback to 6500")
+        return DEFAULT_MIN_MIREDS, DEFAULT_MAX_MIREDS
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
     """Set up the WiZ Platform from config_flow."""
-    # Assign configuration variables.
-    wiz_data = hass.data[DOMAIN][entry.entry_id]
-    wizbulb = WizBulbEntity(wiz_data.bulb, entry.data.get(CONF_NAME), wiz_data.scenes)
-    # Add devices with defined name
-    async_add_entities([wizbulb], update_before_add=True)
-    return True
+    wiz_data: WizData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities([WizBulbEntity(wiz_data, entry.data[CONF_NAME])])
 
 
-class WizBulbEntity(LightEntity):
+class WizBulbEntity(CoordinatorEntity, LightEntity):
     """Representation of WiZ Light bulb."""
 
-    def __init__(self, light: wizlight, name, scenes):
+    def __init__(self, wiz_data: WizData, name: str) -> None:
         """Initialize an WiZLight."""
-        self._light = light
-        self._state = None
-        self._brightness = None
+        super().__init__(wiz_data.coordinator)
+        self._light = wiz_data.bulb
+        bulb_type: BulbType = self._light.bulbtype
+        self._attr_unique_id = self._light.mac
         self._attr_name = name
-        self._rgb_color = None
-        self._temperature = None
-        self._hscolor = None
-        self._available = None
-        self._effect = None
-        self._scenes: list[str] = scenes
-        self._bulbtype: BulbType = light.bulbtype
-        self._mac = light.mac
-        self._attr_unique_id = light.mac
-        # new init states
-        self._attr_min_mireds = self.get_min_mireds()
-        self._attr_max_mireds = self.get_max_mireds()
-        self._attr_supported_features = self.get_supported_features()
+        self._attr_effect_list = wiz_data.scenes
+        self._attr_min_mireds, self._attr_max_mireds = get_min_max_mireds(bulb_type)
+        self._attr_supported_color_modes = get_supported_color_modes(bulb_type)
+        if supports_effects(bulb_type):
+            self._attr_supported_features = SUPPORT_EFFECT
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, self._light.mac)},
+            name=name,
+            manufacturer="WiZ",
+            model=bulb_type.name,
+        )
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if light is on."""
+        is_on: bool | None = self._light.status
+        return is_on
 
     @property
-    def brightness(self):
+    def brightness(self) -> int | None:
         """Return the brightness of the light."""
-        return self._brightness
+        if (brightness := self._light.state.get_brightness()) is None:
```

Comment:
Should we move all these calculations into the `_handle_coordinator_update` callback and just write to `_attr_` properties?

Reply:
Yes.  Planning on doing that in another pass.

Reply:
https://github.com/home-assistant/core/pull/65819